owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt

```diff
@@ -125,114 +127,121 @@ class UsersAndGroupsSearchProvider : ContentProvider() {
-
-            val names = uiResult.data
-
-            // convert the responses from the OC server to the expected format
-            if (!names.isNullOrEmpty()) {
-                response = MatrixCursor(COLUMNS)
-                val namesIt = names.iterator()
-                var item: JSONObject
-                var displayName: String? = null
-                var icon = 0
-                var dataUri: Uri? = null
-                var count = 0
-
-                val userBaseUri = Uri.Builder().scheme(CONTENT).authority(
-                    suggestAuthority!! + DATA_USER_SUFFIX
-                ).build()
-                val groupBaseUri = Uri.Builder().scheme(CONTENT).authority(
-                    suggestAuthority!! + DATA_GROUP_SUFFIX
-                ).build()
-                val remoteBaseUri = Uri.Builder().scheme(CONTENT).authority(
-                    suggestAuthority!! + DATA_REMOTE_SUFFIX
-                ).build()
-
-                val manager = FileDataStorageManager(
-                    context,
-                    account,
-                    context!!.contentResolver
-                )
-                val federatedShareAllowed = manager.getCapability(account!!.name).filesSharingFederationOutgoing
-                    .isTrue
-
-                try {
-                    for (name in names) {
-                        var userName = name.getString(GetRemoteShareesOperation.PROPERTY_LABEL)
-                        val value = name.getJSONObject(GetRemoteShareesOperation.NODE_VALUE)
-                        val type = value.getInt(GetRemoteShareesOperation.PROPERTY_SHARE_TYPE)
-                        val shareWith = value.getString(GetRemoteShareesOperation.PROPERTY_SHARE_WITH)
-
-                        try {
-                            val shareWithAdditionalInfo = value.getString(
-                                GetRemoteShareesOperation.PROPERTY_SHARE_WITH_ADDITIONAL_INFO
-                            )
+        val getShareesResult = getShareesAsyncUseCase.execute(
+            GetShareesAsyncUseCase.Params(
+                userQuery,
+                REQUESTED_PAGE,
+                RESULTS_PER_PAGE
+            )
+        )
 
-                            userName = if (shareWithAdditionalInfo.isEmpty())
-                                userName
-                            else
-                                "$userName ($shareWithAdditionalInfo)"
+        if (getShareesResult.isError) {
+            showErrorMessage(
+                context!!.resources.getString(R.string.get_sharees_error),
+                getShareesResult.getThrowableOrNull()
+            )
+        }
 
-                        } catch (e: JSONException) {
-                            Log_OC.e(TAG, "Exception while parsing shareWithAdditionalInfo", e)
-                        }
+        val names = getShareesResult.getDataOrNull()
+
+        // convert the responses from the OC server to the expected format
+        if (names?.size!! > 0) {
```

Comment:
file translated by Android studio???

Reply:
2 suggestions here:
- Use isNullOrEmpty() function.
- you can remove indentation if you use something like

```
if (names,isNullOrEmpty()) {
    // else block if exists
    return
}
```

Reply:
There's no else condition here so I will go for the first suggestion